### PR TITLE
Use form not div so that enter can submit changes

### PR DESCRIPTION
**Pull request by michaelpacer**
When trying to test edge cases around the keybindings editor (as part of #2153), I found that the inability to hit Enter (<kbd>↲</kbd>) to submit a changed keybinding was a really poor user experience. 

This fixes that; thanks to @Carreau for guidance in figuring this out.

### CODE REVIEW

File: notebook/static/notebook/js/shortcuteditor.js [hunk 1]
```diff
@@ -45,14 +45,19 @@ var KeyBinding = createClass({
...
-    return createElement('div', {className:'jupyter-keybindings'},
+    var binding_setter = function(){
+      if (available) { 
+        that.props.onAddBindings(that.state.shrt, that.props.ckey);
+      }
+      that.state.shrt='';
+      return false;
+    };
+    return createElement('form', {className:'jupyter-keybindings',
+            // onSubmit: function(){ binding_setter(); return false;}
```
Comment (blink1073) [inline notebook/static/notebook/js/shortcuteditor.js]: Extraneous comment

Comment (michaelpacer) [inline notebook/static/notebook/js/shortcuteditor.js]: Oops, fixed.
